scripts/script_utils.py

```diff
@@ -159,3 +161,55 @@ def get_changed_packages(blob_name1, blob_name2, package_list):
             result.add(file_root)
 
     return sorted(result)
+
+
+def get_files_for_linting(allow_limited=True):
+    """Gets a list of files in the repository.
+
+    By default, returns all files via ``git ls-files``. However, in some cases
+    uses a specific commit or branch (a so-called diff base) to compare
+    against for changed files. (This requires ``allow_limited=True``.)
+
+    To speed up linting on Travis pull requests against master, we manually
+    set the diff base to the branch the pull request is against. We don't do
+    this on "push" builds since "master" will be the currently checked out
+    code. One could potentially use ${TRAVIS_COMMIT_RANGE} to find a diff base
+    but this value is not dependable.
+
+    To allow faster local ``tox`` runs, the local remote and local branch
+    environment variables can be set to specify a remote branch to diff
+    against.
+
+    :type allow_limited: bool
+    :param allow_limited: Boolean indicating if a reduced set of files can
+                          be used.
+
+    :rtype: pair
+    :returns: Tuple of the diff base using the the list of filenames to be
```

Comment:
Just noticed "the the" in this docstring. Take this opportunity to fix it? (Almost certainly I am the owner of the original typo.)